src/transaction/log_recovery_analysis.cpp

```diff
@@ -404,11 +404,23 @@ log_recovery_analysis (THREAD_ENTRY *thread_p, INT64 *num_redo_log_records, log_
 	  context.set_start_redo_lsa (start_redo_lsa);
 	}
 
-      log_rv_analysis_record (thread_p, log_rtype, tran_id, &log_nav_lsa, log_page_p, &prev_lsa, context);
-      if (context.is_restore_incomplete ())
+      if (!context.is_page_server ())
```

Comment:
Extract a new - forking function - `log_rv_analysis_record_fork_tran_page` - that does this bit? 
The old forking function can be renaned to `..._tran`. The new page server specific bit - `..._page`.
.. just a thought.

Reply:
Done. Forked `log_rv_analysis_record` into `log_rv_analysis_record_on_page` and `log_rv_analysis_record_on_tran`.